bundles/org.eclipse.equinox.p2.artifact.repository/src/org/eclipse/equinox/internal/p2/artifact/repository/simple/SimpleArtifactRepository.java

```diff
@@ -917,9 +920,15 @@ public IStatus getArtifacts(IArtifactRequest[] requests, IProgressMonitor monito
 			monitor.beginTask(NLS.bind(Messages.sar_downloading, Integer.toString(requests.length)), requests.length);
 			try {
 				DownloadJob jobs[] = new DownloadJob[numberOfJobs];
+				List<IStatus> jobStatus = new CopyOnWriteArrayList<>();
 				for (int i = 0; i < numberOfJobs; i++) {
-					jobs[i] = new DownloadJob(Messages.sar_downloadJobName + i, this, requestsPending, monitor,
-							overallStatus);
+					jobs[i] = new DownloadJob(Messages.sar_downloadJobName + i, this, requestsPending, jobStatus::add,
+							jobMsg -> {
+								synchronized (monitor) {
+									// last message wins
+									monitor.subTask(jobMsg);
+								}
+							});
```

Comment:
If I understand this correctly, then the monitor of the DownloadJob (the `jobMonitor`) now only reports back the task details, but not consume any ticks.

So for example: The "main" job would stay at 10% done, spawn all the "helper" jobs and then jump to 100% once they are finished. Or am I overlooking any other connection between the "parent" and the "worker" progress monitors?

Reply:
Yes, but I think I can account for this as well...

Reply:
Now each completed download is producing one tick for the "parent" monitor.